### DISCUSSION TITLE
Filter streaming

### DISCUSSION THREAD

**Discussion by djrranga** [tweepy/tweepy#1511] [Questions] [answered]
Not sure if this has been answered already.
I'm new to tweepy and have been following the streaming guide on the [docs page](http://docs.tweepy.org/en/latest/streaming_how_to.html). 

This is my entire script (basically copied from the linked page).
```
import tweepy

api_key = ''
api_secret = ''
auth = tweepy.AppAuthHandler(api_key, api_secret)
api = tweepy.API(auth)

class MyStreamListener(tweepy.StreamListener):

    def on_connect(self):
        print("Hi")

    def on_status(self, status):
        print(status.text)

    def on_disconnect(self, notice):
        print(notice)


myStreamListener = MyStreamListener()
myStream = tweepy.Stream(auth=api.auth, listener=myStreamListener)
myStream.filter(track=['python'])
```

There is no output to my console when run, only that it exited with code 0. Any help would be appreciated.

**Comment by Harmon758** [comment]
You should use `StreamListener.on_error` to see if you're encountering any errors.

**Comment by djrranga** [comment]
> You should use `StreamListener.on_error` to see if you're encountering any errors.

Thanks for that suggestion - I'm now seeing 401, which I'm assuming means I'm unauthorized. Based on my code in the original post, can you tell if I've made a mistake in authenticating?

**Comment by Harmon758** [comment, marked as answer]
Ah, you need to use an `OAuthHandler` to authorize with user context for streaming.

**Comment by djrranga** [comment]
Is there a way to authorize with a bearer token (like using a plain GET request with an authentication header)?

**Comment by Harmon758** [comment]
Not for streaming with the [POST statuses/filter endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/filter-realtime/api-reference/post-statuses-filter), no.
It requires authentication with user context.
